### PR TITLE
fix(cloudflare): node_modules chunk

### LINKED ISSUE #211: Cloudflare v10 breaks cloudflare build / wrangler preview with React
### Astro Info

```block
Astro                    v4.5.12
Node                     v20.12.0
System                   Windows (x64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/react
```

### Describe the Bug

When using v10 version of the adapter, hybrid output mode and React on a prerendered path with `client:load` directive, preview with wrangler fails. This was not the case with previous version of the adapter. I haven't tried building the site on Cloudflare, but my guess is if the wrangler fails, then the build will fail too.

Steps to reproduce
- npm install
- npm build
- npm preview

```
Attaching additional modules:
┌──────────────────────────────────────────┬──────┬───────────┐
│ Name                                     │ Type │ Size      │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/astro/assets-service_DrRUv88a.mjs │ esm  │ 10.73 KiB │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/astro_lYPtO5er.mjs                │ esm  │ 83.18 KiB │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/generic_CIq48spH.mjs              │ esm  │ 0.19 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/index_B6euTALA.mjs                │ esm  │ 0.18 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/pages/generic_DCwcQpq8.mjs        │ esm  │ 43.64 KiB │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/pages/index_CPLiUgeG.mjs          │ esm  │ 0.20 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/prerender_Bqre5yrK.mjs            │ esm  │ 0.09 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/test_C85FZqRY.mjs                 │ esm  │ 0.19 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ chunks/vnode-children_DMYleglm.mjs       │ esm  │ 2.95 KiB  │
├──────────────────────────────────────────┼──────┼───────────┤
│ manifest_BqJYjMlt.mjs                    │ esm  │ 18.09 KiB │
├──────────────────────────────────────────┼──────┼───────────┤
│ renderers.mjs                            │ esm  │ 78.01 KiB │
├──────────────────────────────────────────┼──────┼───────────┤
│ _noop-middleware.mjs                     │ esm  │ 0.12 KiB  │
└──────────────────────────────────────────┴──────┴───────────┘
✨ Compiled Worker successfully
 ⛅️ wrangler 3.39.0
-------------------
[wrangler:inf] Ready on http://127.0.0.1:4321
⎔ Starting local server...
X [ERROR] service core:user:issues: Uncaught TypeError: Cannot read properties of undefined (reading 'ReactCurrentDispatcher')

    at null.<anonymous> (vowq86nvp6.js:3366:68) in .wrangler/tmp/pages-Phcscz/renderers.mjs
    at null.<anonymous> (vowq86nvp6.js:4:56) in __init
    at null.<anonymous> (vowq86nvp6.js:7422:1)


X [ERROR] MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.    
```

### What's the expected result?

React should be usable on a prerendered path as it was like this before. If this is not the case, it should be mentioned in the docs that React should be used only with SSR enabled endpoint/path.

### Link to Minimal Reproducible Example

https://github.com/dallyh/astro-cf-10-issues/tree/react-issue

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
## Changes

- This does define a manualChunk to inline `node_modules` according to docs (https://rollupjs.org/configuration-options/#output-manualchunks)
- closes #211 

## Testing

- manual tested on issue repo and exisiting test

## Docs

- bug fix, no docs needed

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
!preview cf-deps-chunk

**Review by natemoo-re — APPROVED** [review]
This makes sense in lieu of a more fundamental fix in core!

File: packages/cloudflare/src/index.ts [hunk 1]
```diff
@@ -209,6 +213,11 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					vite.resolve.conditions = vite.resolve.conditions.filter(
 						(c) => c !== 'workerd' && c !== 'worker'
 					);
+					vite.build ||= {};
+					vite.build.rollupOptions ||= {};
+					vite.build.rollupOptions.output ||= {};
+					//@ts-expect-error
+					vite.build.rollupOptions.output.manualChunks = undefined;
```
Comment (alexanderniebuhr) [inline packages/cloudflare/src/index.ts:220]:
```suggestion
					vite.build.rollupOptions.output.manualChunks ||= undefined;
```

**Comment by alexanderniebuhr** [comment]
!preview cf-deps-chunk

**Comment by alexanderniebuhr** [comment]
!preview cf-deps-chunk

**Review by ematipico — APPROVED** [review]

**Comment by alexanderniebuhr** [comment]
@natemoo-re @ematipico thanks for your reviews, while this is a workaround which works for some cases, it does break for other cases.. I changed to draft again, to investigate further and try to find a better solution, which is more robust, even for a workaround.

**Comment by alexanderniebuhr** [comment]
!preview cf-deps-chunk

File: .changeset/few-beds-dance.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': patch
+---
+
+Fixes chunk issues when bundling node_modules dependencies
```
Comment (alexanderniebuhr) [inline .changeset/few-beds-dance.md:5]: Fixes specific _but not all_ cases where the project can't in Cloudflare, if dependencies from `node_modules` are only used in prerendered pages. This change shouldn't have any side-effects.

**Pull request merged**